glitch_type_client/public/js/main.js

```diff
@@ -0,0 +1,42 @@
+const socket = io()
+
+const { username, room } = Qs.parse(location.search, { ignoreQueryPrefix: true })
+
+let timestamp = 0
+const timeBetweenRefresh = 4
+
+socket.emit("join", { username, room }, (error) => {
+    if (error) {
+        return console.log(error)
+        //redirect to Auth form
+    }
+
+    return console.log(`User ${username} has entered ${room}.`)
+})
+
+socket.emit("start", id, timestamp = (new Date()).getTime(), (error) => {
+    if (error) {
+        return console.log(error)
+    }
+    return console.log(`UserId ${id} found!`)
+})
+
+socket.on("startGameMessage", (info) => {
+    RoomTimestamp = info.timestamp
+    room = info.room
+    for (let i = 0; ; i++) {
+        while (true) {
+            if ((new Date()).getTime() >= info.timestamp + (timeBetweenRefresh * 1000) * i) {
+                //Send Scores for user
+                score = 0
+                //"score" should be the present score of the client.
+                socket.emit("recentScore", score.info)
+                break;
+            }
+        }
+    }
+})
+
+socket.on("leaderboard", (leaderboard) => {
+
+})
```

Comment:
What is this file for?

Reply:
Every time a new player joins the game in a specific room the scores of every other player is updated and rendered ,to maintain a sync between players sending the updated score a timestamp is related to every room which is equal to the timestamp of when the last player joined the room. The for loop adds 4000ms to the timestamp related to that room and while loop keeps the condition to be repeatedly checked and once the condition gets fulfilled the client emits the score to be updated in the server. The for loop never ends as the score must keep on updating as long as the player is in the game and once the socket gets closed it will automatically break

Reply:
You were not supposed to make changes or additions to the client side code. You kinda put it in the wrong place. I'm merging the PR anyway because I'll just pick up from your code and port it to React, but this file in itself won't be used. Just letting you know.